### PR TITLE
Included a few tests, doc and fixed error codes

### PR DESCRIPTION
**Pull request by chandrams**
Made these changes:
- Included a doc for remote monitoring tests
- Fixed error codes and updated mandatory fields test for create experiment
- Added a test in listRecommendations to test latest withtout experiment name

### CODE REVIEW

File: tests/scripts/remote_monitoring_tests/remote_monitoring_tests.sh [hunk 1]
```diff
@@ -37,13 +37,7 @@ function remote_monitoring_tests() {
 	((TOTAL_TEST_SUITES++))
 
 	python3 --version >/dev/null 2>/dev/null
-	err=$?
-	if [ ${err} -ne 0 ]; then
-		echo "ERROR: python3 not installed"
-		exit 1
-	fi
-	
-	python3 -m pip install --user -r "${REMOTE_MONITORING_TEST_DIR}/requirements.txt" >/dev/null 2>&1
```
Comment (dinogun) [inline tests/scripts/remote_monitoring_tests/remote_monitoring_tests.sh:46]: Is it intentional to remove this install package line as well?

File: tests/scripts/remote_monitoring_tests/remote_monitoring_tests.sh [hunk 2]
```diff
@@ -111,6 +112,8 @@ function remote_monitoring_tests() {
 		pushd ${REMOTE_MONITORING_TEST_DIR}/rest_apis > /dev/null 
 			echo "pytest -m ${test} --html=${TEST_DIR}/report.html --cluster_type ${cluster_type}"
 			pytest -m ${test} --html=${TEST_DIR}/report.html --cluster_type ${cluster_type} | tee -a ${LOG}
+			err_exit "ERROR: Running the test using pytest failed, check ${LOG} for details!"
```
Comment (dinogun) [inline tests/scripts/remote_monitoring_tests/remote_monitoring_tests.sh:115]: We seem to be using `check_err` in some places and `err_exit` in others.

**Comment by dinogun** [comment]
@chandrams it looks like none of the commits are signed, can you please ensure all your commits are signed.

File: tests/scripts/remote_monitoring_tests/remote_monitoring_tests.sh (hunk 1 above)
Comment (chandrams) [reply tests/scripts/remote_monitoring_tests/remote_monitoring_tests.sh:46]: I have moved it below.

**Comment by chandrams** [comment]
@dinogun - Yes, have issues with the gpg keys on my new laptop.

**Pull request merged**